### PR TITLE
ERA5 on-the-fly CMORizer: changed sign of `evspsbl` and `evspsblpot`

### LINKED ISSUE #2116: Dataset problem: In ERA5 evspsbl and evspsblpot have the opposite sign compared to CMOR
**Describe the dataset issue**
A clear and concise description of what the problem with the data is is. Here are some guidelines that you can use
right out the box by filling in the blanks (note that, if needed we will assist you with writing a fix; also note that
if the problem is with a CMIP6 dataset, we will alert the ESGF and/or the model developers for them to fix the problem
in the long run; fixes for CMIP5 by the model developers are no longer possible):

For ERA5 the evapotranspiration and potential evapotranspiration are negative, if they go from the surface into the atmosphere for CMIP models they are positive:
According to the description of Evapotranspiration on the Copernicus page (https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means?tab=overview):
> The ECMWF Integrated Forecasting System (IFS) convention is that downward fluxes are positive. Therefore, negative values indicate evaporation and positive values indicate condensation.

In the CMOR table, these fluxes are defined as positive, if they go from the surface into the atmosphere:
> Evaporation at surface (also known as evapotranspiration): flux of water into the atmosphere due to conversion of both liquid and solid phases to vapor (from underlying surface and vegetation)

- Data file has been changed by you in any way (if you answer yes the issue will be void and closed, we are not
supporting custom-made datasets that present problems, it is your resposability to fix those problems): no
- Project (CMIP5/CMIP6/OBS/obs4MIPs/etc): native6
- Full dataset description (fill out as many as you know, please):
  - dataset: ERA5
  - experiment: 
  - mip: Amon
  - grid: 
  - type: reanalysis
  - version: v1
  - tier: 3 
  - variable used: evspsbl and evspsblpot
  - data version:
- Problems encountered (please describe what the actual problems are: e.g. wrong standard name, issue with dimensions): issue with sign
- Pointer to existing copy of data on ESGF node (it would be very useful if you could provide a physical
fullpath to the file(s) that are causing the problem, e.g. on CEDA Jasmin or DKRZ):
DKRZ: 
/work/bd0854/DATA/ESMValTool2/RAWOBS/Tier3/ERA5/v1/mon/evspsblpot/*
and
/work/bd0854/DATA/ESMValTool2/RAWOBS/Tier3/ERA5/v1/mon/evspsbl/*
- Other notes and mentions:

**Assign the right people**
If you are already a member of the ESMValTool GitHub project, please assign Valeriu Predoi (valeriupredoi) and
Klaus Zimmermann (zklaus) to the issue. They will then check the issue raised and propagate it further to the
data model developers.

**Please attach**
  - The recipe that you are trying to run, you can find a copy in the `run` directory in the output directory
  - The `main_log_debug.txt` file, this can also be found in the `run` directory in the output directory
  - Attach a text file with the output of `ncdump -h /path/to/file.nc`

Cheers :beer:

### PR DESCRIPTION
**Pull request by katjaweigel**
Change sign of evspsbl and evspsblpot to have the same direction as CMIP models. Added both variables for ERA5_land.

## Description
For ERA5 evapotranspiration and potential evapotranspiration are negative, if they go from the surface into the atmosphere for CMIP models they are positive:

According to the description of Evapotranspiration on the Copernicus page (https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means?tab=overview): 
> The ECMWF Integrated Forecasting System (IFS) convention is that downward fluxes are positive. Therefore, negative values indicate evaporation and positive values indicate condensation.

In the CMOR table, these fluxes are defined as positive, if they go from the surface into the atmosphere:
>  Evaporation at surface (also known as evapotranspiration): flux of water into the atmosphere due to conversion of both liquid and solid phases to vapor (from underlying surface and vegetation)

Therefore the fix to cmorize the ERA5 (and ERA5-Land) data needs to switch their sign.

Note that the sign of these variables would change, everywhere where they are used. So far, I did not find any plot or calculation based on these variables from ERA5, but they are included in 
ESMValTool/esmvaltool/recipes/cmorizers/recipe_daily_era5.yml
and
ESMValTool/esmvaltool/recipes/examples/recipe_check_obs.yml
and could have been used to create data to use outside ESMValTool diagnostics.

Closes #2116

Link to documentation: https://github.com/ESMValGroup/ESMValCore/blob/fix_era5_evspsbl/doc/quickstart/find_data.rst

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [ ] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by katjaweigel** [comment]
This is not ready, but I accidentally pressed the wrong button, when I wanted to make a draft PR.

**Comment by katjaweigel** [comment]
It seems to fail on hourly data - I don't understand if there is really an issue with them (could be, because there is a time dependent unit conversion involved, but that has actually not changed) or if it does not have the hourly data (we don't have them on Levante but they exist on the Copernicus page)?

**Comment by rswamina** [comment]
@valeriupredoi - Calling your attention to this as requested in the morning session of the workshop.

**Comment by valeriupredoi** [comment]
@katjaweigel it was just a test needing fixing for the two evspbl hourly variables - if the expected data before fixing is negative then the test data has to be negative as well :grin:

**Comment by katjaweigel** [comment]
@valeriupredoi thanks a lot, I'd have never found that test.

**Comment by valeriupredoi** [comment]
> @valeriupredoi thanks a lot, I'd have never found that test.

my pleasure, am the chief mechanic after all :grin:

**Review by valeriupredoi — APPROVED** [review]
many thanks @katjaweigel :beer:

File: esmvalcore/cmor/_fixes/native6/era5.py [hunk 1]
```diff
@@ -111,6 +110,7 @@ def fix_metadata(self, cubes):
             fix_hourly_time_coordinate(cube)
             fix_accumulated_units(cube)
             multiply_with_density(cube)
+            cube.data = cube.core_data() * -1.0
```
Comment (remi-kazeroni) [inline esmvalcore/cmor/_fixes/native6/era5.py:113]: Could you maybe add a one-line comment below "Set input cube units for invalid units were ignored on load" for the change of sign? Maybe:
`# Correct sign to align with CMOR standards`
(and maybe same for the second variable fixed)

File: doc/quickstart/find_data.rst [hunk 2]
```diff
@@ -121,6 +121,16 @@ ERA5
...
 
+.. note:: According to the description of Evapotranspiration and potential Evapotranspiration on the Copernicus page 
+  (https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means?tab=overview):
+  "The ECMWF Integrated Forecasting System (IFS) convention is that downward fluxes are positive. 
+  Therefore, negative values indicate evaporation and positive values indicate condensation."
+  
+  In the CMOR table, these fluxes are defined as positive, if they go from the surface into the atmosphere:
+  "Evaporation at surface (also known as evapotranspiration): flux of water into the atmosphere due to conversion 
+  of both liquid and solid phases to vapor (from underlying surface and vegetation)."
+  Therefore the fix to cmorize the ERA5 (and ERA5-Land) switches their sign to be comparable to the CMIP models.
```
Comment (remi-kazeroni) [inline doc/quickstart/find_data.rst:132]: Maybe something like:
```suggestion
  Therefore, the ERA5 (and ERA5-Land) CMORizer switches the signs of ``evspsbl`` and ``evspsblpot`` to be comparable to the CMIP models.
```

**Review by remi-kazeroni — COMMENTED** [review]
Thanks, @katjaweigel! Looks good to me. I just have a couple of minor suggestions for the documentation. I have edited the title of the PR to mention ERA5, feel free to revert back of you prefer.

 I have also removed the "backward incompatible" label because this PR corrects a bug in a CMORization for variables not following CMOR standards. I think that label is not really needed here. Question: do we have recipes in `main` that would be affected by this change? If so, could you please list them in the description of this PR?

File: doc/quickstart/find_data.rst (hunk 2 above)
Comment (katjaweigel) [reply doc/quickstart/find_data.rst]: ok, thanks!

**Comment by katjaweigel** [comment]
Dear @remi-kazeroni thanks for checking! I included your comments. the only recipes in main, which contain these variables from ERA5 I found are:
ESMValTool/esmvaltool/recipes/examples/recipe_check_obs.yml and 
ESMValTool/esmvaltool/recipes/cmorizers/recipe_daily_era5.yml
I included that in the description above.

File: doc/quickstart/find_data.rst [hunk 3]
```diff
@@ -121,6 +121,16 @@ ERA5
...
 
+.. note:: According to the description of Evapotranspiration and potential Evapotranspiration on the Copernicus page 
+  (https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means?tab=overview):
+  "The ECMWF Integrated Forecasting System (IFS) convention is that downward fluxes are positive. 
+  Therefore, negative values indicate evaporation and positive values indicate condensation."
+  
+  In the CMOR table, these fluxes are defined as positive, if they go from the surface into the atmosphere:
+  "Evaporation at surface (also known as evapotranspiration): flux of water into the atmosphere due to conversion 
+  of both liquid and solid phases to vapor (from underlying surface and vegetation)."
+  Therefore, the ERA5 (and ERA5-Land) CMORizer switches the signs of ``evspsbl`` and ``evspsblpot`` to be comparable to the CMIP models.
```
Comment (axel-lauer) [inline doc/quickstart/find_data.rst:132]:
```suggestion
  Therefore, the ERA5 (and ERA5-Land) CMORizer switches the signs of ``evspsbl`` and ``evspsblpot`` to be compatible with the CMOR standard used e.g. by the CMIP models.
```

**Review by axel-lauer — APPROVED** [review]
Looks good to me. Thanks fot spotting and fixing this! Maybe consider changing the wording in line 132 of `find_data.rst` e.g. as suggested.

**Comment by katjaweigel** [comment]
@axel-lauer thanks a lot and sorry that I forgot that change when I included the other ones.

**Pull request merged**